doc/scripts/gen_state_diagram.py

```diff
@@ -107,7 +107,7 @@ def get_state(self, line):
                 r"boost::statechart::state_machine<\s*(\w*),\s*(\w*)\s*>",
```

Comment:
changing L93 to `match = re.search(r"(\w+::)*::(?P<tag>\w+)::\w+\(\s*const (?P<event>\w+)",` makes the script pass locally for me